torch/_inductor/ir.py

```diff
@@ -1252,8 +1253,21 @@ def _dynamic_reshape_indexer(old_size, new_size):
         size_hint = V.graph.sizevars.size_hint
         vars = [sympy_symbol(f"view{i}") for i in range(len(new_size))]
 
-        stack_new = list(zip(vars, new_size))
-        stack_old = list(old_size)
+        def any0(l):
+            return any([size_hint(x) for x in list(l)])
+
+        if any0(old_size) or any0(new_size):
+            assert any0(old_size) and any0(new_size)
+            view_expr = [sympy.Integer(0)] * len(old_size)
+            stack_new = []
+            stack_old = []
+            numel = math.prod(vars, start=sympy.Integer(1))
```

Comment:
I'm pretty sure this should be `sympy_product`, and `vars` is placeholder variable you just created, so you cannot guard on it, it's pointless

Reply:
yes, I should be guarding on the old_size vars, I mixed up the two